functions/controllers/application/index.js

```diff
@@ -127,4 +127,14 @@ application.post("/submit", jwtCheck, hasUpdateApp, async (req, res) => {
   }
 });
 
+application.get("/checkApp", jwtCheck, hasReadApp, async (req, res) => {
+  try {
+    doc = await queryDocument("applicants", req.user.sub);
+    appStatus = doc.get("status");
+    res.status(200).send({ code: 200, status: appStatus, exists: true, message: "Document Found" });
```

Comment:
I ran some more tests. This will not throw an error for failing it's just going to return null and thus will always return 200. The logic here will need to check if that is doc is empty or not. I'd recommend running these tests again manually with the emulator

Reply:
I think you need to either check the appStatus value that's it's valid and throw an error in this situation.